src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3612,21 +3612,27 @@ void handleMemberInitializer(IMemberInitializerOperation memberInitializer)
                         // the value has been evaluated. So we assemble the reference after visiting the value.
 
                         _evalStack.Push(Visit(memberReference.Instance));
-                        return (success: true, propertyArguments);
-
+                        return (success: true, propertyArguments.CastArray<IOperation>());
```

Comment:
Use `CastUp` instead of `CastArray`. The former has compile time verification this is a safe up cast.